protos/idl/service.proto

```diff
@@ -98,3 +109,68 @@ message Tag {
 message Metadata {
     map<string, string> key_map = 1; // key map is a dictionary of key/val strings that represent metadata
 }
+
+message FilterExpression {
+    repeated SingleValueFilter filters = 1;
+}
+
+message SingleValueFilter {
+    oneof entity {
+        TagFilter tagFilter = 1;
+        PartitionFilter partition_filter = 2;
+        ArtifactFilter artifact_filter = 3;
+        DatasetFilter dataset_filter = 4;
+    }
+
+    // as use-cases come up we can add more operators, ex: gte, like, not eq etc.
+    enum ComparisonOperator {
+        EQUALS = 0;
+    }
+
+    ComparisonOperator operator = 10; // field 10 in case we add more entities to query
+    // Next field number: 11
+}
+
+message ArtifactFilter {
+    oneof property {
+        string artifact_id = 1;
```

Comment:
Isn't an artifact_id only unique within the dataset it's in? As in, the artifact_id is not unique across all datasets within a domain?

Reply:
At the moment, the artifact_id is unique across all datasets. I think all clients are passing uuid as the artifact_id right now. Do you envision that as a problem?

Reply:
No, that would actually be preferred. We currently query for specific artifacts by specifying both the artifact type (since we have created a hierarchy in which each artifact type has its own dataset) and an artifact_id. If we were to be able to just pass in an artifact_id and could search the whole Catalog Service for it (probably with a project, domain, and version filter), that would be excellent!